owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoveRemoteShareOperation.kt

```diff
@@ -46,63 +43,59 @@
+class RemoveRemoteShareOperation
+/**
+ * Constructor
+ *
+ * @param remoteShareId Share ID
+ */
+    (private val mRemoteShareId: Long) : RemoteOperation<ShareParserResult>() {
 
-    @Override
-    protected RemoteOperationResult run(OwnCloudClient client) {
-        RemoteOperationResult result;
+    override fun run(client: OwnCloudClient): RemoteOperationResult<ShareParserResult> {
+        var result: RemoteOperationResult<ShareParserResult>
 
         try {
-            Uri requestUri = client.getBaseUri();
-            Uri.Builder uriBuilder = requestUri.buildUpon();
-            uriBuilder.appendEncodedPath(ShareUtils.SHARING_API_PATH);
-            uriBuilder.appendEncodedPath(String.valueOf(mRemoteShareId));
+            val requestUri = client.baseUri
+            val uriBuilder = requestUri.buildUpon()
+            uriBuilder.appendEncodedPath(ShareUtils.SHARING_API_PATH)
+            uriBuilder.appendEncodedPath(mRemoteShareId.toString())
 
-            DeleteMethod deleteMethod = new DeleteMethod(
-                    new URL(uriBuilder.build().toString())
-            );
+            val deleteMethod = DeleteMethod(
+                URL(uriBuilder.build().toString())
+            )
 
-            deleteMethod.addRequestHeader(OCS_API_HEADER, OCS_API_HEADER_VALUE);
+            deleteMethod.addRequestHeader(RemoteOperation.OCS_API_HEADER, RemoteOperation.OCS_API_HEADER_VALUE)
 
-            int status = client.executeHttpMethod(deleteMethod);
+            val status = client.executeHttpMethod(deleteMethod)
 
             if (isSuccess(status)) {
 
                 // Parse xml response and obtain the list of shares
-                ShareToRemoteOperationResultParser parser = new ShareToRemoteOperationResultParser(
-                        new ShareXMLParser()
-                );
-                result = parser.parse(deleteMethod.getResponseBodyAsString());
+                val parser = ShareToRemoteOperationResultParser(
+                    ShareXMLParser()
+                )
+                result = parser.parse(deleteMethod.responseBodyAsString)
 
-                Log_OC.d(TAG, "Unshare " + mRemoteShareId + ": " + result.getLogMessage());
+                Log_OC.d(TAG, "Unshare " + mRemoteShareId + ": " + result.logMessage)
 
             } else {
-                result = new RemoteOperationResult<>(deleteMethod);
+                result = RemoteOperationResult(deleteMethod)
             }
 
-        } catch (Exception e) {
-            result = new RemoteOperationResult<>(e);
-            Log_OC.e(TAG, "Unshare Link Exception " + result.getLogMessage(), e);
+        } catch (e: Exception) {
+            result = RemoteOperationResult(e)
+            Log_OC.e(TAG, "Unshare Link Exception " + result.logMessage, e)
         }
 
-        return result;
+        return result
     }
 
-    private boolean isSuccess(int status) {
-        return (status == HttpConstants.HTTP_OK);
+    private fun isSuccess(status: Int): Boolean {
```

Comment:
you can refactor this function in a single line

rivate fun isSuccess(status: Int): Boolean = status == HttpConstants.HTTP_OK

Reply:
👍